src/app/plugins/gamble.plugin.ts

```diff
@@ -0,0 +1,116 @@
+import { MessageEmbed } from 'discord.js';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, Maybe } from '../../common/types';
+import { PointsDocument, PointsModel } from '../../schemas/points.schema';
+
+export default class GamblePlugin extends Plugin {
+  public commandName: string = 'gamble';
+  public name: string = 'Gamble Plugin';
+  public description: string = 'Bet your points on a coin flip';
+  public usage: string = 'gamble 100\ngamble all';
+  public override pluginAlias = [];
+  public permission: ChannelType = ChannelType.All;
+
+  public override commandPattern: RegExp = /(all|\d+)/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const userDoc: Maybe<PointsDocument> = await PointsModel.findOne({
```

Comment:
Is there a find or create

Reply:
Don't think so
But I making something similar to that inside of a `container.PointService`